funfact/initializers.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from abc import ABC, abstractmethod
+from numbers import Number
+from funfact import active_backend as ab
+
+
+class _Initializer(ABC):
+
+    @abstractmethod
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def __call__(self, *args, **kwargs):
+        return type(self)(*args, **kwargs)
+
+    @abstractmethod
+    def init(self, *args, **kwargs):
```

Comment:
It might indeed be a bit confusing, but the good news is that users are not expected to see or use this method at all. They only need to use the initializer object as an opaque, function-like object.

That said, we can rename it to `_init` or `_gen` or something along the line to avoid confusion,

Reply:
Done by simplifying the initializers structure.